### ISSUE TITLE
Docker recommended docker stetup

### ISSUE THREAD

**Issue by olymk2** [pythonic-emacs/anaconda-mode#326]
Something I keep struggling with is docker and this setup can you recommend how to set this up ?

Currently I have .dir-locals setup with this param.

             (python-shell-interpreter . "/usr/local/bin/python3")

The problem with this is that it uses my local file unless I open a file with in the container seems there should be a nicer way.

Can I not use and edit the files locally but make it so that everything is run against the copy in the container they are of course the same files so seems some kind of mapping would need to happen is there a way to do this .

basically I want to be able to open up my local python project and start working but get auto complete and repl and everything else from the container any tips ?

**Comment by proofit404** [comment]
Hi,

There is an experimental docker-compose support in the pythonic library https://github.com/proofit404/pythonic/blob/master/pythonic.el#L154

How to try it: 

1. Add this to your config

```lisp
(add-hook 'python-mode-hook 'pythonic-set-docker-compose-alias)
(setq python-shell-interpreter "/usr/local/bin/python")
```

2. Open any python file inside your project with docker-compose.yml

3. Anaconda-mode and djangonaut will be able to work with remote interpreter in the local folder.

Please, give it a try!

Regards, Artem.

**Comment by olymk2** [comment]
Okay that's awesome, I will give it a try and let you know may be a week or two now but I definitely want to try that, and I can see you can override the docker-compose.yml as I tend to have a local.yml for local dev.

**Comment by olymk2** [comment]
Initial testing looks good, switched to project it picked up my local.yml from dir-local and asked me for a container to use selected django from the list and tried running djangonaut at which point it connected to the container and gave me a list of the management commands.

Very slick I will test this more extensively and let you know if I find any issues, thanks for making thi integration making emacs python development a lot nicer for me :)

**Comment by proofit404** [comment]
I'm happy to see that things useful for me are useful for others.

**Comment by CeleritasCelery** [comment]
Looks like this is addressed
